src/kudu/rpc/client_negotiation.cc

```diff
@@ -166,6 +166,16 @@ Status ClientNegotiation::Negotiate(unique_ptr<ErrorStatusPB>* rpc_error) {
   // Ensure we can use blocking calls on the socket during negotiation.
   RETURN_NOT_OK(CheckInBlockingMode(socket_.get()));
 
+  // Step 0: Perform normal TLS handshake if enabled
+  if (tls_context_->GetEnableNormalTLS()) {
```

Comment:
so what happens if the server is not enabled for normal TLS?

Reply:
Added use_normal_tls to gate client behavior. On the other hand, enable_normal_tls gates both the client and server.